service/interceptor/interceptor.go

```diff
@@ -0,0 +1,232 @@
+		rpcConn, err := connection.Connect(retrieverAddress, metricsManager, connection.OnConnectionLoss(connection.ExitOnConnectionLoss()))
+		if err != nil {
+			log.Error(err.Error())
+		}
+
+		retrieverClient := retriever.NewMetadataRetrieverClient(rpcConn, 100*time.Second)
+		if retrieverClient == nil {
+			log.Error("Cannot get csi-metadata-retriever client")
+		}
+
+		i.opts.MetadataSidecarClient = retrieverClient
+	} else {
+		log.Warn("env var not found: ", "CSI_RETRIEVER_ENDPOINT")
+	}
+}
+
+const pending = "pending"
+
+func (i *interceptor) nodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRequest,
+	info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (res interface{}, resErr error) {
+	lock, err := i.opts.locker.GetLockWithID(ctx, req.VolumeId)
+	if err != nil {
+		return nil, err
+	}
+
+	if closer, ok := lock.(io.Closer); ok {
+		defer closer.Close() // #nosec G307
+	}
+
+	if !lock.TryLock(i.opts.timeout) {
+		return nil, status.Error(codes.Aborted, pending)
+	}
+	defer lock.Unlock()
+
+	return handler(ctx, req)
+}
+
+func (i *interceptor) nodeUnstageVolume(ctx context.Context, req *csi.NodeUnstageVolumeRequest,
+	info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (res interface{}, resErr error) {
+	lock, err := i.opts.locker.GetLockWithID(ctx, req.VolumeId)
+	if err != nil {
+		return nil, err
+	}
+	if closer, ok := lock.(io.Closer); ok {
+		defer closer.Close() // #nosec G307
+	}
+	if !lock.TryLock(i.opts.timeout) {
+		return nil, status.Error(codes.Aborted, pending)
+	}
+	defer lock.Unlock()
+
+	return handler(ctx, req)
+}
+
+func (i *interceptor) createVolume(ctx context.Context, req *csi.CreateVolumeRequest,
+	info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (res interface{}, resErr error) {
+
+	lock, err := i.opts.locker.GetLockWithID(ctx, req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	if closer, ok := lock.(io.Closer); ok {
+		defer closer.Close() // #nosec G307
+	}
+
+	if !lock.TryLock(i.opts.timeout) {
+		return nil, status.Error(codes.Aborted, pending)
+	}
+	defer lock.Unlock()
+
+	// This should be corrected to SoftLimt changes.
+	metadataReq := &retriever.GetPVCLabelsRequest{
+		Name:      req.Parameters["csi.storage.k8s.io/pvc/pvcsoftlimit"],
+		NameSpace: req.Parameters["csi.storage.k8s.io/pvc/pvcsoftlimit"],
+		// This should be corrected to SoftLimt changes.
+	}
+
+	if i.opts.MetadataSidecarClient != nil {
+		metadataRes, err := i.opts.MetadataSidecarClient.GetPVCLabels(ctx, metadataReq)
+		if err != nil {
+			/*			log.Errorf("Cannot retrieve labels for PVC %s in namespace: %s, error: %v",
+						controller.KeyCSIPVCName,
+						controller.KeyCSIPVCNamespace,
+						err.Error())
+			*/
+		}
+
+		if metadataRes != nil {
+			for k, v := range metadataRes.Parameters {
+				log.Info("***Start: Meta Data From PVC Yaml File**")
+				log.Info(k)
+				log.Info(v)
+				log.Info("***End: Meta Data From PVC Yaml File**")
+				req.Parameters[k] = v
+			}
+		} else {
```

Comment:
else condition is empty , please check this

Reply:
it is logging a warning. not empty